### ISSUE TITLE
SSD Mobilenet is not detecting any object. 

### ISSUE THREAD

**Issue by senthilnathansss** [fdchiu/SNPECam#3]
Compiled and ran app in the Pixel-2 but no objects are getting detected.
I know the coco trained SSD mobilenet works for Person and some objects like cup, monitor but this app detects nothing.

**Comment by fdchiu** [comment]
It should detect some objects. Are you using your own model or the model in
the repo? Another thing to check is the camera, make sure the image is fed
to ssd mobile net is correct image (size/exposure/etc).

David

**Comment by unitedsoft** [comment]
I tracked it looks like the fix is to change the below IF condition in run method of SNAPNet.java class. 

 //if (output.getKey().equals(mOutputLayer)) {
            if (key.startsWith("Postprocessor/BatchMultiClassNonMaxSuppression")) {


- SP

**Comment by fdchiu** [comment]
Thanks for the update!
Everyone's network is different from others. IN your case your out put is
"Postprocessor/BatchMultiClassNonMaxSuppression"  so this change worked for
you.

Just wondering if you set mOutputLayer to the above value, should the
original statement still work?

David

**Comment by unitedsoft** [comment]
Yes, you are right. Setting mOutputLayer should work. But haven't tested it yet.
But thank again your work helped me and saved lot of time for me.

-- SP

**Comment by jeranio** [comment]
Hello

I was not getting any detections at all until I reversed the commented line as so:

```java
//if (output.getKey().equals(mOutputLayer)) {
if (key.startsWith("Postprocessor/BatchMultiClassNonMaxSuppression")) {
```

the orginal file was 

```java
if (output.getKey().equals(mOutputLayer)) {
//if (key.startsWith("Postprocessor/BatchMultiClassNonMaxSuppression")) {
```

Then it starts to detect but it detects all kinds of things not even in the room like elephants. Is there a guide on training this model to be more accuate.

Thanks though for the good work

Running on SDM660 platform
